app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationService.kt

```diff
@@ -0,0 +1,791 @@
+            .setSmallIcon(R.drawable.ic_notify)
+            .setContentIntent(eventResultPendingIntent)
+            .setColor(context.getColor(R.color.notification_color))
+            .setAutoCancel(true)
+            .setShortcutId(account.id.toString())
+            .setSubText(account.fullName)
+            .setVisibility(NotificationCompat.VISIBILITY_PRIVATE)
+            .setCategory(NotificationCompat.CATEGORY_SOCIAL)
+            .setOnlyAlertOnce(true)
+            .setGroup(channelId)
+            .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_SUMMARY) // Only ever alert for the summary notification
+
+        setSoundVibrationLight(account, builder)
+
+        return builder
+    }
+
+    private fun titleForType(notification: Notification, account: AccountEntity): String? {
+        if (notification.status == null) {
+            return null
+        }
+
+        val accountName = notification.account.name.unicodeWrap()
+        when (notification.type) {
+            Notification.Type.MENTION -> return context.getString(R.string.notification_mention_format, accountName)
+            Notification.Type.STATUS -> return context.getString(R.string.notification_subscription_format, accountName)
+            Notification.Type.FOLLOW -> return context.getString(R.string.notification_follow_format, accountName)
+            Notification.Type.FOLLOW_REQUEST -> return context.getString(R.string.notification_follow_request_format, accountName)
+            Notification.Type.FAVOURITE -> return context.getString(R.string.notification_favourite_format, accountName)
+            Notification.Type.REBLOG -> return context.getString(R.string.notification_reblog_format, accountName)
+            Notification.Type.POLL -> return if (notification.status.account.id == account.accountId) {
+                context.getString(R.string.poll_ended_created)
+            } else {
+                context.getString(R.string.poll_ended_voted)
+            }
+            Notification.Type.SIGN_UP -> return context.getString(R.string.notification_sign_up_format, accountName)
+            Notification.Type.UPDATE -> return context.getString(R.string.notification_update_format, accountName)
+            Notification.Type.REPORT -> return context.getString(R.string.notification_report_format, account.domain)
+            Notification.Type.UNKNOWN -> return null
+        }
+    }
+
+    private fun bodyForType(notification: Notification, alwaysOpenSpoiler: Boolean): String? {
+        if (notification.status == null) {
+            return null
+        }
+
+        when (notification.type) {
+            Notification.Type.FOLLOW, Notification.Type.FOLLOW_REQUEST, Notification.Type.SIGN_UP -> return "@" + notification.account.username
+            Notification.Type.MENTION, Notification.Type.FAVOURITE, Notification.Type.REBLOG, Notification.Type.STATUS -> return if (!TextUtils.isEmpty(notification.status!!.spoilerText) && !alwaysOpenSpoiler) {
```

Comment:
the `!!` is unnecessary as you checked already for null above

Reply:
Well spotted (already removed).